signs-work-order-map/components/CalcWidthLengthComponent.tsx

```diff
@@ -0,0 +1,90 @@
+import Form from "react-bootstrap/Form";
+import Container from "react-bootstrap/Container";
+import { Col, Row } from "react-bootstrap";
+import { Controller } from "react-hook-form";
+import ControlledNumberInput from "@/components/ControlledNumberInput";
+import {
+  MAXFEET,
+  MAXWIDTH,
+  MAXTHICKNESS,
+  useCalculation,
+} from "@/config/calcs";
+import { CalcWidthLengthComponentProps } from "@/types/calcs";
+
+export default function CalcWidthLengthComponent({
+  title,
+  control,
+  watch,
+  setValue,
+  inputWidthName,
+  inputLengthName,
+  inputThicknessName,
+  outputName,
+  outputLabel,
+  calculationName,
+}: CalcWidthLengthComponentProps) {
+  const [widthField, lengthField] = watch([inputWidthName, inputLengthName]);
+
+  let thicknessField = undefined;
+
+  if (inputThicknessName) {
+    [thicknessField] = watch([inputThicknessName]);
+  }
+
+  useCalculation({
+    width: widthField,
+    length: lengthField,
+    thickness: thicknessField,
+    calculationName: calculationName,
+    setValue,
+    outputFieldName: outputName,
+  });
+
+  return (
+    <Container className="m-3 p-3">
+      <Row>
+        <h4>{title}</h4>
+      </Row>
+      <Row>
+        <Col>
```

Comment:
i don't know how these different forms are used, but one way to clean this up visually would be to wrap each section in a `Card` component and move the title into the `Card.Title`. The output going into the `Card.Footer` could help as well 🤔